zilencer/views.py

```diff
@@ -714,7 +715,33 @@ def remote_server_post_analytics(
             )
         batch_create_table_data(server, RemoteRealmAuditLog, remote_realm_audit_logs)
 
-    return json_success(request)
+    remote_realm_dict: Dict[str, Any] = {}
+    remote_realms = RemoteRealm.objects.filter(server=server)
+    for remote_realm in remote_realms:
+        uuid = str(remote_realm.uuid)
+        remote_realm_dict[uuid] = {
+            "can_push": False,
+            "expected_end_timestamp": None,
+        }
+
+        billing_session = RemoteRealmBillingSession(remote_realm)
+        customer = billing_session.get_customer()
+        if customer is None:
+            continue
+        current_plan = get_current_plan_by_customer(customer)
+        if current_plan:
+            expected_end_timestamp = None
+            if current_plan.status in [
+                CustomerPlan.DOWNGRADE_AT_END_OF_CYCLE,
+                CustomerPlan.DOWNGRADE_AT_END_OF_FREE_TRIAL,
+            ]:
+                expected_end_timestamp = billing_session.get_next_billing_cycle(current_plan)
+            remote_realm_dict[uuid] = {
+                "can_push": True,
+                "expected_end_timestamp": expected_end_timestamp,
+            }
```

Comment:
Let's use the early return pattern here, i.e. have after `if customer is None` an `if current_plan is None` block that fills a value in and continues...

Reply:
Done